### PR TITLE
Resolve http redirect

### PR DESCRIPTION
**Pull request by su-chang**
@windmemory pls confirm, thx!

### CODE REVIEW

File: src/gateway/wechat-gateway.ts [hunk 1]
```diff
@@ -335,7 +335,20 @@ export class WechatGateway extends EventEmitter {
         const rawData: any = []
         let dataLen = 0
 
-        if (response.statusCode !== 200) {
+        if (response.statusCode === 302) {
+          const location = response.headers.location || res.commandUrl
```
Comment (windmemory) [inline src/gateway/wechat-gateway.ts]: Let's put some log here for investigation purpose, log out the redirected location here in info level. Also let's add some code here to deal with situation that status code is `301`, which is permanent changed location, log out the location for `301` status code.

File: src/gateway/wechat-gateway.ts [hunk 2]
```diff
@@ -335,7 +335,26 @@ export class WechatGateway extends EventEmitter {
...
+          log.info(PRE, '302 location', location)
+          let hostname: string | undefined
+          if (/^(http|https):\/\//.test(location)) {
+            const hostnameReg = new RegExp(/(?<=\/\/).*?(?=\/)/)
+            const hostnameBody = location.match(hostnameReg)
+            hostname = hostnameBody ? hostnameBody[0] : location
+            log.info(PRE, 'hostname', hostname)
```
Comment (windmemory) [inline src/gateway/wechat-gateway.ts]: This is not necessary.

File: src/gateway/wechat-gateway.ts [hunk 3]
```diff
@@ -335,7 +335,26 @@ export class WechatGateway extends EventEmitter {
...
+            const pathReg = new RegExp('/(?<=' + hostname + ').*/')
+            const path = location.match(pathReg)![0]
+            res.commandUrl = path
+            log.info(PRE, 'path', path)
```
Comment (windmemory) [inline src/gateway/wechat-gateway.ts]: ditto

File: src/gateway/wechat-gateway.ts (hunk 2 above)
Comment (su-chang) [reply src/gateway/wechat-gateway.ts]: Add this log to check the REG is right or not, maybe I should use log.debug() instead log.info()?

Comment (windmemory) [reply src/gateway/wechat-gateway.ts]: Yep, you might need `log.silly()`

Comment (windmemory) [reply src/gateway/wechat-gateway.ts]: `log.info()` should only log out informations that our user needed. And for this case, we log out the `location` for temporary investigation purpose, after we solve this problem entirely, we should remove the info log of location.

**Comment by su-chang** [comment]
Sorry, I forget how to make it back to `0.3.17`, just modify in `package.json`?

**Comment by windmemory** [comment]
That should be good.

**Pull request merged**